src/command.rs

```diff
@@ -54,9 +54,17 @@ impl Command {
       Self::Operation(operation) => state.operation = *operation,
       Self::Print => state.print()?,
       Self::Repl => {
-        for result in BufReader::new(io::stdin()).lines() {
-          let line = result?;
-          match line.trim().parse::<Self>() {
+        let history = tilde("~/.degenerate_history").to_string();
+
+        let mut rl = Editor::<()>::new();
```

Comment:
I'd call it editor, since it's a little bit clearer:

```suggestion
        let mut editor = Editor::<()>::new();
```